erpnext/hr/doctype/employee_compensation/employee_compensation.py

```diff
@@ -3,8 +3,20 @@
 # For license information, please see license.txt
 
 from __future__ import unicode_literals
-# import frappe
+import frappe
+from frappe import _
+from frappe.utils import today
 from frappe.model.document import Document
 
 class EmployeeCompensation(Document):
-	pass
+	def validate(self):
+		self.validate_dates()
+
+	def validate_dates(self):
+		today_date = today()
+		if self.date_of_injury and self.date_of_injury > today_date:
+			frappe.throw(_("You cannot select a future date for Date of Injury."))
+		if self.first_aid_date and self.first_aid_date > today_date:
+			frappe.throw(_("You cannot select a future date for First Aid Date."))
```

Comment:
```suggestion
		iif (self.date_of_injury or self.first_aid_date) and (self.date_of_injury or self.first_aid_date) > today_date:
			frappe.throw(_("You cannot select a future date for Date of Injury / First Aid Date."))
```

Reply:
@vishdha let it be separate, combining them won't work for different use cases.